### PR TITLE
fixes #6227

### LINKED ISSUE #6227: DuckDBJ: DuckDBDatabaseMetaData.getTables always throws when specifying a catalog argument, whereas DuckDBDatabaseMetaData.getColumns doesn't
### What happens?

`java.sql.DatabaseMetadata.getTables()` supports a `catalog` argument (https://docs.oracle.com/en/java/javase/11/docs/api/java.sql/java/sql/DatabaseMetaData.html#getTables(java.lang.String,java.lang.String,java.lang.String,java.lang.String%5B%5D)). However, the implementation in `org.duckdb.DuckDBDatabaseMetaData`  (https://github.com/duckdb/duckdb/blob/master/tools/jdbc/src/main/java/org/duckdb/DuckDBDatabaseMetaData.java#L676) always throws when it is not null or empty:

```
		if (catalog != null && !catalog.isEmpty()) {
			throw new SQLException("Actual catalog argument is not supported, got " + catalog);
		}        
```

It could be that duckdb simply chooses not to support this, but then it is strange (inconsistent) that a method like `org.duckdb.DuckDBDatabaseMetaData.getColumns` handles the catalog argument just like any other argument (i.e. it just uses it to filter and does not throw when not empty or null)

(BTW there is another issue with getColumns which is related but still different https://github.com/duckdb/duckdb/issues/6225)

### To Reproduce

	public static void test_get_tables_with_catalog() throws Exception {
		Connection conn = DriverManager.getConnection("jdbc:duckdb:");
		String catalog = conn.getCatalog();
		DatabaseMetaData databaseMetaData = conn.getMetaData();
		try {
			databaseMetaData.getTables(catalog, null, "%", null);
		}
		catch (SQLException ex) {
			assertFalse(ex.getMessage().startsWith("Actual catalog argument is not supported"));
		}
	}

### OS:

Windows 10 Pro

### DuckDB Version:

v0.6.2-dev2342

### DuckDB Client:

DuckDBJ

### Full Name:

Roland Bouman

### Affiliation:

Just-Bi.nl (an EPAM company) 

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by rpbouman**
- adds a test that demonstrates old implementation of getTables always throws an SQLException
- rewrites getTables to provide handling for the catalog parameter.

### CODE REVIEW

**Review by Mytherin — APPROVED** [review]
Thanks for the PR! LGTM. Perhaps @Mause can also have a look?

File: tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java [hunk 1]
```diff
@@ -1821,6 +1821,18 @@ public static void test_schema_reflection() throws Exception {
 
 		conn.close();
 	}
+	
+	public static void test_get_tables_with_catalog() throws Exception {
+		Connection conn = DriverManager.getConnection("jdbc:duckdb:");
+		String catalog = conn.getCatalog();
+		DatabaseMetaData databaseMetaData = conn.getMetaData();
```
Comment (Mytherin) [inline tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java:1828]: Could we add some tests that verify the `getTables` correctly takes the catalog into account, by e.g. attaching an in-memory database and creating tables there?

Comment (rpbouman) [reply tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java:1828]: Sure! Will pick it up tonight.

Comment (rpbouman) [reply tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java:1829]: @Mytherin - something like this? https://github.com/duckdb/duckdb/pull/6230/commits/df0303c652e2b874ec5005b7faf712e80f47dee4

**Comment by Mytherin** [comment]
Looks good - thanks!

**Pull request merged**